Source/Scene/ModelExperimental/ModelExperimental.js

```diff
@@ -0,0 +1,369 @@
+ * <br /><br />
+ * If this object was destroyed, it should not be used; calling any function other than
+ * <code>isDestroyed</code> will result in a {@link DeveloperError} exception.
+ *
+ * @returns {Boolean} <code>true</code> if this object was destroyed; otherwise, <code>false</code>.
+ *
+ * @see ModelExperimental#destroy
+ */
+ModelExperimental.prototype.isDestroyed = function () {
+  return false;
+};
+
+/**
+ * Destroys the WebGL resources held by this object.  Destroying an object allows for deterministic
+ * release of WebGL resources, instead of relying on the garbage collector to destroy this object.
+ * <br /><br />
+ * Once an object is destroyed, it should not be used; calling any function other than
+ * <code>isDestroyed</code> will result in a {@link DeveloperError} exception.  Therefore,
+ * assign the return value (<code>undefined</code>) to the object as done in the example.
+ *
+ * @exception {DeveloperError} This object was destroyed, i.e., destroy() was called.
+ *
+ *
+ * @example
+ * model = model && model.destroy();
+ *
+ * @see ModelExperimental#isDestroyed
+ */
+ModelExperimental.prototype.destroy = function () {
+  var gltfLoader = this._gltfLoader;
+  if (defined(gltfLoader)) {
+    gltfLoader.destroy();
+  }
+
+  var resources = this._resources;
+  for (var i = 0; i < resources.length; i++) {
+    resources[i].destroy();
+  }
+
+  destroyObject(this);
+};
+
+function initialize(model, options) {
+  var gltf = options.gltf;
+
+  var loaderOptions = {
+    baseResource: options.basePath,
+    releaseGltfJson: options.releaseGltfJson,
+    incrementallyLoadTextures: options.incrementallyLoadTextures,
+  };
+
+  if (gltf instanceof Uint8Array) {
+    loaderOptions.typedArray = gltf;
+    loaderOptions.gltfResource = Resource.createIfNeeded(
+      defaultValue(options.basePath, "")
+    );
+  } else {
+    loaderOptions.gltfResource = gltf;
+  }
+
+  var loader = new GltfLoader(loaderOptions);
+  model._gltfLoader = loader;
+  loader.load();
+
+  loader.promise
+    .then(function (loader) {
+      model._sceneGraph = new ModelExperimentalSceneGraph({
+        model: model,
+        modelComponents: loader._components,
+        upAxis: options.upAxis,
+        forwardAxis: options.forwardAxis,
+        modelMatrix: options.modelMatrix,
+      });
+      model._resourcesLoaded = true;
+    })
+    .otherwise(function () {
+      ModelExperimentalUtility.getFailedLoadFunction(
+        this,
+        "model",
+        options.basePath
+      );
+    });
+
+  loader.texturesLoadedPromise
+    .then(function () {
+      model._texturesLoaded = true;
+    })
+    .otherwise(function () {
+      ModelExperimentalUtility.getFailedLoadFunction(
+        this,
+        "model",
+        options.basePath
+      );
+    });
+}
+
+/**
+ *
+ * @param {Object} options Object with the following properties:
+ * @param {Resource|String} options.url The url to the .gltf or .glb file.
```

Comment:
Since we have already decided to make a breaking change, I thought it's ok to make a parameter called `gltf` that can handle all cases:
 - `String` - glTF Path
 - `Resource` - glTF Path
 - `Uint8Array` - glb Typed Array
We'll add a PR to add support for loading glTF JSON.